### DISCUSSION TITLE
Allow GoogleAdsToGcsOperator use search_proto_plus() instead of search() method

### DISCUSSION THREAD

**Discussion by ag-q** [apache/airflow#29932] [Ideas]
There is no way to use `search_proto_plus()` instead of default `search()` method in _airflow/providers/google/ads/transfers/ads_to_gcs.py_ without overriding the existing operator.

I mean specifically this line:
`rows = service.search(client_ids=self.client_ids, query=self.query, page_size=self.page_size)`

The operator should allow you to choose the appropriate method between search() and search_proto_plus(), because there are enum fields in Google Ads that cannot be imported as strings (names) without the search_proto_plus() method.

For now I managed by overriding the default `search()` method with `search_proto_plus()`, but it would be great to have the option to choose between these methods in the package.

Apache Airflow version: 2.4.3
apache-airflow-providers-google==8.8.0

**Comment by potiuk** [comment]
Feel free to contribute it - no issue needed, PR is more than enough.
